src/backend/utils/mmgr/aset.c

```diff
@@ -1109,6 +1109,7 @@ AllocSetDelete(MemoryContext context)
 		AllocBlock	next = block->next;
 		size_t freesz = UserPtr_GetUserPtrSize(block);
         MemoryContextNoteFree(&set->header, freesz);
+		MemoryContextNoteFree(parent, freesz);
```

Comment:
why not
```c
AllocSet pset = (AllocSet) parent;
AssertArg(AllocSetIsValid(pset));
MemoryContextNoteFree(&pset->header, freesz);
```

Reply:
May be is it worth to inject `MemoryContextNoteFree()` call for parent context into `MemoryContextDeleteImpl()` before recursive `MemoryContextDeleteChildren()` call?

Reply:
smth like this:
```C
diff --git a/src/backend/utils/mmgr/aset.c b/src/backend/utils/mmgr/aset.c
index 4cde8a874c..26739403af 100644
--- a/src/backend/utils/mmgr/aset.c
+++ b/src/backend/utils/mmgr/aset.c
@@ -1107,8 +1107,6 @@ AllocSetDelete(MemoryContext context)
     while (block != NULL)
     {
         AllocBlock  next = block->next;
-        size_t freesz = UserPtr_GetUserPtrSize(block);
-        MemoryContextNoteFree(&set->header, freesz);
 
 #ifdef CLOBBER_FREED_MEMORY
         wipe_mem(block, block->freeptr - ((char *) block));
diff --git a/src/backend/utils/mmgr/mcxt.c b/src/backend/utils/mmgr/mcxt.c
index de22f54005..a614c77ed5 100644
--- a/src/backend/utils/mmgr/mcxt.c
+++ b/src/backend/utils/mmgr/mcxt.c
@@ -228,6 +228,8 @@ MemoryContextResetChildren(MemoryContext context)
 void
 MemoryContextDeleteImpl(MemoryContext context, const char* sfile, const char *func, int sline)
 {
+    size_t freesz = context->allBytesAlloc - context->allBytesFreed;
+
     AssertArg(MemoryContextIsValid(context));
     /* We had better not be deleting TopMemoryContext ... */
     Assert(context != TopMemoryContext);
@@ -239,6 +241,7 @@ MemoryContextDeleteImpl(MemoryContext context, const char* sfile, const char *fu
     context->callerLine = sline;
 #endif
 
+    MemoryContextNoteFree(context->parent, freesz);
     MemoryContextDeleteChildren(context);
 
     /*
```